### PR TITLE
[API][sims-#517]  restrict institution Admin from accessing/ editing locations or location related features of other institutions

### PR DESCRIPTION
**Pull request by ann-aot**
1. Restrict institution Admin from accessing/ editing locations or location related features of other institutions
2.  Restrict institution Admin from accessing/ editing/ creating offerings, and programs related features of other institutions

### CODE REVIEW

File: sources/packages/api/src/route-controllers/education-program-offering/education-program-offering.controller.ts
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/education-program-offering/education-program-offering.controller.ts:63]: At this point, the institution access was already validated, I would recommend a throw a NotFoundException.

File: sources/packages/api/src/route-controllers/education-program-offering/education-program-offering.controller.ts
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/education-program-offering/education-program-offering.controller.ts:54]: The fixes in this way look good to complete this ticket, but I believe that we should move this check inside the InstitutionLocationGuard. What is happening now is that we are not checking the locations for admins.
To fix it in the InstitutionLocationGuard, I believe that we should.

1. Load the locations Ids when the user is an Admin on JwtStrategy
2. Remove the below code from InstitutionLocationGuard and let the location ids be validated for Admin also.
```
if (authorizations.isAdmin()) {
   return true;
}
```

File: sources/packages/api/src/services/institution-location/institution-location.service.ts
Comment (andrewsignori-aot) [inline sources/packages/api/src/services/institution-location/institution-location.service.ts:38]: This could be simplified as below:
```
.where("institution_location.id = :id", { id })
```

File: sources/packages/api/src/services/institution-location/institution-location.service.ts
Comment (andrewsignori-aot) [inline sources/packages/api/src/services/institution-location/institution-location.service.ts:37]: It seems that there is no need to return the entire institution record here. Please consider returning only what is need for now.
I believe that you can also use the leftJoinAndSelect, if you think that it would be better.

**Review by abschwenker — APPROVED** [review]
LGTM, once reviews are done please merge

**Review by andrewsignori-aot — APPROVED** [review]
👍

**Review by guru-aot — APPROVED** [review]
LGTM

File: sources/packages/api/src/route-controllers/education-program-offering/education-program-offering.controller.ts
Comment (dheepak-aot) [inline sources/packages/api/src/route-controllers/education-program-offering/education-program-offering.controller.ts:54]: Code from line 54-58 has been repeated in multiple places. 
Hence this piece of code can be added to a method and called. It will reduce the redundant code.

**Pull request merged**
